### PR TITLE
feat(sast): Split sast and cdk reports

### PR DESCRIPTION
**Pull request by achiar99**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

1) In case we run sast+cdk frameworks - run sast-core only once and get merged report
2) Split the report to sast + cdk reports

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/bridgecrew/platform_integration.py [hunk 1]
```diff
@@ -601,6 +601,23 @@ def persist_sast_scan_results(self, reports: List[Report]) -> None:
 
         persist_checks_results(sast_scan_reports, self.s3_client, self.bucket, self.repo_path)  # type: ignore
 
+    def persist_cdk_scan_results(self, reports: List[Report]) -> None:
+        cdk_scan_reports = {}
+        for report in reports:
+            if not report.check_type.startswith('cdk'):
```
Comment (tronxd) [inline checkov/common/bridgecrew/platform_integration.py:607]: use `CDK_FRAMEWORK_PREFIX`

File: checkov/sast/engines/prisma_engine.py [hunk 2]
```diff
@@ -351,8 +356,83 @@ def create_report(self, prisma_report: PrismaReport) -> List[SastReport]:
...
+                for policy_id, rule_match in rule_matches.items():
+                    if rule_match.metadata.framework != 'CDK':
+                        sast_rule_matches[policy_id] = rule_match
+                        continue
+                    if lang not in [c.language for c in cdk_reports]:
+                        new_cdk_report = PrismaReport(rule_match={lang: {}}, errors=report.sast_report.errors,
+                                                  profiler=report.sast_report.profiler,
+                                                  run_metadata=report.sast_report.run_metadata,
+                                                  imports={}, reachability_report={})
+                        new_report = CDKReport(f'cdk_{lang.value}', report.sast_report.run_metadata, lang, new_cdk_report)
```
Comment (tronxd) [inline checkov/sast/engines/prisma_engine.py:377]: use `CDK_FRAMEWORK_PREFIX`

File: checkov/common/runners/runner_registry.py [hunk 3]
```diff
@@ -195,6 +199,28 @@ def run(
                                                         self.runners if runner.resource_subgraph_map is not None}
         return self.scan_reports
 
+    def _merge_runners(self, runners: list[_BaseRunner]) -> list[_BaseRunner]:
```
Comment (tronxd) [inline checkov/common/runners/runner_registry.py:202]: add UT to this function

File: checkov/sast/checks_infra/base_registry.py [hunk 4]
```diff
@@ -12,6 +12,8 @@
 logger = logging.getLogger(__name__)
 
 
+CDK_CHECKS_DIR_PATH = f'{os.getcwd()}/checkov/cdk/checks'
```
Comment (tronxd) [inline checkov/sast/checks_infra/base_registry.py:15]: use `CDK_FRAMEWORK_PREFIX`

File: checkov/sast/engines/prisma_engine.py [hunk 5]
```diff
@@ -351,8 +356,83 @@ def create_report(self, prisma_report: PrismaReport) -> List[SastReport]:
...
+        all_report = self.split_sast_cdk_reports(reports)
+        return all_report
+    
+    def split_sast_cdk_reports(self, sast_reports: List[SastReport]) -> List[Union[SastReport, CDKReport]]:
+        cdk_reports: List[CDKReport] = []
+        for report in sast_reports:
+            for lang, rule_matches in report.sast_report.rule_match.items():
+                sast_rule_matches: Dict[str, RuleMatch] = {}
+                for policy_id, rule_match in rule_matches.items():
+                    if rule_match.metadata.framework != 'CDK':
```
Comment (tronxd) [inline checkov/sast/engines/prisma_engine.py:369]: why is `CDK` upper case here?

File: checkov/sast/engines/prisma_engine.py [hunk 6]
```diff
@@ -351,8 +356,83 @@ def create_report(self, prisma_report: PrismaReport) -> List[SastReport]:
                 report = SastReport(f'{self.check_type.lower()}_{lang.value}', prisma_report.run_metadata, lang, sast_report)
                 report.sast_reachability = prisma_report.reachability_report[lang]
                 reports.append(report)
-        return reports
 
+        all_report = self.split_sast_cdk_reports(reports)
+        return all_report
+    
+    def split_sast_cdk_reports(self, sast_reports: List[SastReport]) -> List[Union[SastReport, CDKReport]]:
```
Comment (tronxd) [inline checkov/sast/engines/prisma_engine.py:363]: is it possible to split the function into smaller ones?

File: checkov/sast/engines/prisma_engine.py [hunk 7]
```diff
@@ -351,8 +356,83 @@ def create_report(self, prisma_report: PrismaReport) -> List[SastReport]:
...
+                break
+
+        for skip_check in report.skipped_checks:
+            for cdk_report in cdk_reports:
+                if report.language == cdk_report.language and skip_check.check_id not in [s.check_id for s in cdk_report.skipped_checks]:
+                    sast_skiped_checks.append(skip_check)
+                    break
+            else:
+                sast_skiped_checks = report.skipped_checks
+                break
```
Comment (tronxd) [inline checkov/sast/engines/prisma_engine.py:418]: the failed/skipped logic looks the same, put it into a function that iterates on a check lists

File: checkov/cdk/runner.py [hunk 8]
```diff
@@ -31,6 +32,8 @@ def run(
     ) -> list[Report]:
         runner_filter = runner_filter or RunnerFilter()
         runner_filter.remove_default_sast_policies = True
+        # TODO - add more langs
```
Comment (pazbechor) [inline checkov/cdk/runner.py:35]: Why not adding all languages?

File: checkov/common/bridgecrew/platform_integration.py [hunk 9]
```diff
@@ -601,6 +601,23 @@ def persist_sast_scan_results(self, reports: List[Report]) -> None:
...
+                continue
+            if not report.cdk_report:  # type: ignore
+                continue
+            for _, match_by_check in report.cdk_report.rule_match.items():  # type: ignore
```
Comment (pazbechor) [inline checkov/common/bridgecrew/platform_integration.py:611]: iterate values?

**Review by pazbechor — APPROVED** [review]

**Review by tronxd — APPROVED** [review]

**Pull request merged**